backend/matchmaker_service.py

```diff
@@ -0,0 +1,170 @@
+json_config_path = _TEST_FIRESTORE_INSTANCE_JSON_PATH
+_AMAZON_SQS_PROD_CONF_JSON_PATH = '../amazon/stopthevirus.fifo.json'
+_TEST_AMAZON_SQS_CONFIG_PATH = '../amazon/stopthevirus.fifo.json'
+_TEST_TWILIO_SMS_CONFIG_PATH = '../twilio/stv-twilio.json'
+
+
+
+class MatchmakerService:
+    # Handles scheduling and communication with other services for starting games
+    # TDOO(David): Add function to run all games that are supposed to be running at start(in MVP/test)
+    def __init__(self, matchmaker: MatchMakerInterface, gamedb: Database, json_config_path=json_config_path, region="US", min_players=5, is_mvp=True):
+        self._matchmaker = matchmaker
+        self._gamedb = gamedb
+        self._min_players = min_players
+        self._region = region
+        self._is_mvp = is_mvp
+        self._stop = threading.Event()
+        self._daemon_started = False
+
+    def play_game(self, game: Game, players:list, game_dict:dict, is_test=True):
+        log_message("Starting a game", game_id=game_dict.get("id"), additional_tags=game_dict)
+
+        if is_test:
+            database = MockDatabase()
+            engine = MockPlayEngine().CreateEngine(database)
+        else:
+            database = FirestoreDB(json_config_path=json_config_path, game_id=game._game_id)#db needs to have correct game_id
+            engine = Engine(options=game._options,
+                            game_id=game._game_id,
+                            sqs_config_path=_TEST_AMAZON_SQS_CONFIG_PATH,
+                            twilio_config_path=_TEST_TWILIO_SMS_CONFIG_PATH,
+                            gamedb=database
+            )
+
+        game_data = self._matchmaker.generate_teams_tribes(game_id=game._game_id, players=players, game_options=game._options, gamedb=database)
+        tribes = game_data['tribes']
+        # TODO notify users that game started 
+        game.play(tribe1=tribes[0],
+                tribe2=tribes[1],
+                gamedb=database,
+                engine=engine)
+
+
+    def start_game(self, game: Game, game_snap: DocumentSnapshot, players:list, game_dict: dict):
+        self.set_game_has_started(game_snap=game_snap, game=game)
+        if self._is_mvp:
+            #start new process
+            p = multiprocessing.Process(target=self.play_game, args=(game, players, game_dict))
+            p.start()
+        else:
+            #start on new GCP instance
+            pass
+
+    def set_game_has_started(self, game_snap: DocumentSnapshot, game: Game):
+        field_updates = {
+            'game_has_started': True
+        }
+        try:
+            game_snap.reference.update(field_updates)
+            log_message(message="Marked game as started in db", game_id=game._game_id)
+        except Exception as e:
+            log_message(message="Error setting game document game_has_started field to True: {}".format(e), game_id=game._game_id)
+    
+    def reschedule_cancel_game(self, game_snap: DocumentSnapshot, game_dict: dict):

```

Comment:
I think this name is a little confusing. is this the reschedule or the cancel state in the service model diagram?

Reply:
This does both. I'm not sure if there's a convention for naming in this case or if I should split this up into 2 separate functions